### DISCUSSION TITLE
Can I use an entities sortKey as number

### DISCUSSION THREAD

**Discussion by obiwabrakenobi** [typedorm/typedorm#183] [Q&A] [answered]
I have an entity which needs as sortKey a number

`@Entity({
    name: 'Event',
    primaryKey: {
        partitionKey: 'test___{{id}}',
        sortKey: '{{timestamp}}', // timestamp is number
    },
})`

timestamp is of type number but as far as I can see the sortKey timestamp is then rendered as string before PUT. Is it possible to change that?

**Comment by whimzyLive** [comment, marked as answer]
Hi @obiwabrakenobi ,

Yes, it is supported via `aliases`.
if your entity contains an attribute named as `timestamp`, then instead of defining a template for sortKey as `{{timestamp}}`, you can  directly reference that attribute, like

```
@Entity<Event>({
  name: 'Event',
  ...,
  sortKey: {
    alias: 'timestamp'   // <- this tells TypeDORM to auto infer "type" and "value" for partition key from "timestamp" attribute.
    }
})
class Event {
@Attribute()
timestamp: number 
}

```

Let me know if this helps.

**Reply by obiwabrakenobi** [reply]
I have verion `1.14.6` installed and sortKey is only a `string`? I can't find it in here as well 

https://github.com/typedorm/typedorm/blob/main/packages/common/src/metadata/metadata-storage.ts

Am I missing something?

**Reply by whimzyLive** [reply]
@obiwabrakenobi 

Yes, It is available since 1.13.0.

However, to utilize the attribute aliasing feature, you will need to provide an entity as a generic entity, for the type guards to correctly resolve allowed attribute names.

It will need to look something like (notice the `<Event>` generic in the below example)
```
@Entity<Event>({
  name: 'Event',
  ...,
  sortKey: {
    alias: 'timestamp'   // <- this tells TypeDORM to auto infer "type" and "value" for partition key from "timestamp" attribute.
    }
})
class Event {
@Attribute()
timestamp: number 
}
```

P.S. I have also updated the above answer with the correct example.

**Reply by obiwabrakenobi** [reply]
Yes even the generic method does not work for me. Are the typescript defintions wrong here? I cant find it

If I cast it, it works:

```typescript
@Entity<Event>({
    name: 'Event',
    primaryKey: {
        partitionKey: 'test___{{id}}',
        sortKey: {
            alias: 'timestamp',
        } as unknown as string,
    },
})
```

As fast as I can see the generic don't put anything onto the `PrimaryKey`

```typescript
export interface EntityRawMetadataOptions<Entity = any> {
    name: string;
    target: EntityTarget<Entity>;
    primaryKey: PrimaryKey;
    indexes?: Indexes<Entity>;
    table?: Table;
}
```

**Reply by whimzyLive** [reply]
@obiwabrakenobi,


It's definitely a bug. I will get it fixed in the next release.

Looks like the implementation was updated to handle aliases in the primary key, but type declaration is still referring to the old implementation. 

Thanks for reporting.

**Reply by whimzyLive** [reply]
Update: This is now tracked as #194
